### PR TITLE
Day16 Ex01 TabLyout & ViewPager

### PR DESCRIPTION
**Pull request by at-datbui**
#### Related issues (Optional)
Task #84
#### Description (Required)
- [x] Custom bottom tablayout when selected
- [x] Change color text and icon in tablayout when selected

Create a ViewPagerA with 3 tabs ( each item is a fragment )
- [x] Disable buffer loading of viewpager
- [x] Custom TabLayout ( Image attached )

Create a ViewPagerB for slide images ( on tab 2 of A ):
- [x] Duration when change slide = 5s
#### Reference (Optional)

### CODE REVIEW

File: app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        mViewPager.setPageTransformer(true, new ZoomOutPageTransformer());
+        run();
+        changeDurationScroll();
+        return view;
+    }
+
+    private void run() {
+        // Auto start of viewpager
+        final Handler handler = new Handler();
+        final Runnable Update = new Runnable() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java]: ERROR: Name 'Update' must match pattern '^[a-z][a-zA-Z0-9]*$'.

File: app/src/main/java/vn/asiantech/internship/exday16/PageAFragment.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_page_a, container, false);
+        mImages = new ArrayList<>();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/PageAFragment.java]: ERROR: [STYLE][ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD] Write to static field vn.asiantech.internship.exday16.PageAFragment.mImages from instance method vn.asiantech.internship.exday16.PageAFragment.onCreateView(LayoutInflater, ViewGroup, Bundle)
http://findbugs.sourceforge.net/bugDescriptions.html#ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD

File: app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_page_b, container, false);
+        mImages = new ArrayList<>();
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java]: ERROR: [STYLE][ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD] Write to static field vn.asiantech.internship.exday16.PageBFragment.mImages from instance method vn.asiantech.internship.exday16.PageBFragment.onCreateView(LayoutInflater, ViewGroup, Bundle)
http://findbugs.sourceforge.net/bugDescriptions.html#ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD

File: app/src/main/java/vn/asiantech/internship/exday16/PageAFragment.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 28-06-2017.
+ */
+public class PageAFragment extends Fragment {
+    private ViewPager mViewPager;
+    private TabLayout mTabLayout;
+    private final String[] mTextTab = {"Tab 1", "Tab 2", "Tab 3", "Tab 4", "Tab 5"};
+    private static final int[] mTabIconsSelected = {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/PageAFragment.java]: ERROR: Name 'mTabIconsSelected' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.

File: app/src/main/java/vn/asiantech/internship/exday16/ItemImageBFragment.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 28-06-2017.
+ */
+public class ItemImageBFragment extends Fragment {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/ItemImageBFragment.java]: ERROR: [PMD-CPD] 70 lines duplicated.
app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java

File: app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.List;
+
+import vn.asiantech.internship.R;
+
+/**
+ * Copyright © 2016 AsianTech inc.
+ * Created by datbu on 28-06-2017.
+ */
+public class ItemImageFragment extends Fragment {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java]: ERROR: [PMD-CPD] 70 lines duplicated.
app/src/main/java/vn/asiantech/internship/exday16/ItemImageBFragment.java

File: app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            @Override
+            public void run() {
+                handler.post(update);
+            }
+        }, 1000, 5000);
+    }
+
+    private void changeDurationScroll() {
+        try {
+            Field mScroller;
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java]: `mScroller` to `scroller`

File: app/src/main/java/vn/asiantech/internship/exday16/ViewPagerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+    ViewPagerAdapter(FragmentManager fm, List<String> images) {
+        super(fm);
+        mImages = images;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        if (position == 1) {
+            return new PageBFragment();
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/internship/exday16/ViewPagerAdapter.java]: please create `newInstance()` method for `PageBFragment` class

File: app/src/main/res/layout/activity_viewpager.xml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <FrameLayout
+        android:id="@+id/fragmentViewPager"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_viewpager.xml]: id to `frameViewPager`

File: app/src/main/res/layout/fragment_page_a.xml [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <android.support.v4.view.ViewPager
+        android:id="@+id/container"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/fragment_page_a.xml]: `container` to `viewPager`

File: app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    private ImageView mImageView;
+    private boolean mIsStarted ;
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java]: ERROR: ';' is preceded with whitespace.

File: app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+    private boolean mIsVisible ;
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/ItemImageFragment.java]: ERROR: ';' is preceded with whitespace.

File: app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+    private void run() {
+        // Auto start of viewpager
+        final Handler handler = new Handler();
+        final Runnable update = new Runnable() {
+            public void run() {
+                boolean isLoop = false;
+                if (mCurrentPage == mViewPagerAdapter.getCount()) {
+                    mCurrentPage = -1;
+                    isLoop = !isLoop;
+                    if (isLoop == true && mCurrentPage == -1) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/internship/exday16/PageBFragment.java]: ERROR: Expression can be simplified.

**Comment by at-vinhhuynh** [comment]
@at-datbui 
Please resolve conflict

**Comment by at-datbui** [comment]
Thanks @at-vinhhuynh review for me!

**Comment by at-tienhoang** [comment]
@at-datbui Please resolve conflict

**Pull request merged**